### PR TITLE
Cleanup src/coreclr/src/inc/src/metadata.h

### PR DESCRIPTION
**Pull request by sdmaclea**
Remove #include of private implementation headers
Fix build breaks

### CODE REVIEW

**Comment by sdmaclea** [comment]
I was a bit inconsistent with the added `#include` lines.  I am happy to change them.  I just couldn't decide whether to use "_file_" or <_file_>

File: src/coreclr/src/inc/sstring.h [hunk 1]
```diff
@@ -46,6 +46,10 @@
 #include "utilcode.h"
 #include "sbuffer.h"
 
+#ifndef _ASSERTE
+#define _ASSERTE(x)
```
Comment (jkotas) [inline src/coreclr/src/inc/sstring.h:50]: Would it be better to include debugmacros.h instead?

IIRC, we had cases where code like this disabled asserts in large parts of the code and allowed bugs to sneak through.

**Review by jkotas — APPROVED** [review]

File: src/coreclr/src/inc/sstring.h (hunk 1 above)
Comment (sdmaclea) [reply src/coreclr/src/inc/sstring.h:50]: I would love to add the right piece of code.  I didn't think `debugmacros.h` was working.  I eventually gave up as I saw a similar pattern in other files.

I'll try it again.

Comment (sdmaclea) [reply src/coreclr/src/inc/sstring.h]: Done

**Pull request merged**